src/RescriptCore.res

```diff
@@ -68,3 +68,13 @@ type undefined<+'a> = Js.undefined<'a>
 type nullable<+'a> = Js.nullable<'a>
 
 let panic = Core__Error.panic
+
+let protect = (~finally, f) => {
+  let result = try f() catch {
+  | exn =>
+    finally()
+    raise(exn)
```

Comment:
I know it looks like some heresy, but this way, it will have a better compiled code without wrapping Js errors into JsError exception.
```suggestion
  | _ =>
    finally()
    raise(%raw(`exn`))
```

Reply:
Doesn't seem to be any wrapping as-is either: https://rescript-lang.org/try?version=v10.1.2&code=DYUwLgBADgTg9mEBjSBeCAKAfgMwJYB2AhsMAJ4A0EOAlBKgHwQDeAUBBKJDCAM4CuwNBDAwy1DHSREwSABYt2EAD4QQADwL0GSjvmKkyk3RBhE8vEBg0EaSgL5L9JcsY48BQ1o6A

Reply:
Oh, nice. Looks like that's because the compiler knows that `raise` accepts `exn` or whatever.
https://rescript-lang.org/try?version=v10.1.2&code=DYUwLgBADgTg9mEBjSBeCAKAfgMwJYB2AhsMAJ4A0EOAlBKgHwQDeAUBBKJDCAM4CuwNBDAwy1DHSREwSABYt2EAD4QQADwL0GSjvmKkyk3RABSvAHTA4AcwwaCNEzCJ5eIe5qccAvkv0k5MYcPAJCrH5AA